src/PublishPort.jsx

```diff
@@ -7,10 +8,47 @@ import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput"
 import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
 import { OutlinedQuestionCircleIcon, TrashIcon } from '@patternfly/react-icons';
 import cockpit from 'cockpit';
+import ipaddr from "ipaddr.js";
+
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import * as utils from './util.js';
 
 const _ = cockpit.gettext;
 
-export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
+export function validatePublishPort(value, key, validationFailed) {
+    const delta = { ...validationFailed };
+
+    switch (key) {
+    case "IP":
+        if (value && !ipaddr.isValid(value))
+            delta.IP = _("Must be a valid IP address");
+        break;
+    case "hostPort": {
+        if (!utils.isEmpty(value)) {
+            const hostPort = parseInt(value);
+            if (hostPort < 1 || hostPort > 65536)
+                delta.hostPort = _("1 to 65536");
+        }
+
+        break;
+    }
+    case "containerPort": {
+        if (utils.isEmpty(value))
+            delta.containerPort = _("Container port must not be empty");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231009-115441-607719f9-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)